.github/workflows/ci-standard-checks.yml

```diff
@@ -0,0 +1,28 @@
+name: CI Standard Checks
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  ci-standard-checks:
+    # for private repositories
+    runs-on: [self-hosted, automated-checks]
+    # for public repositories
+    # runs-on: [ubuntu-latest]
+
+    steps:
+      - name: Check Out Source Code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
Should we try to fix this "clone everything"? We had this same issue with the secrets scans. Here's what other people with the same problem did [How to get the full range of commits for a PR](https://github.community/t/how-to-get-the-full-range-of-commits-for-a-pr/17527), but of course we can explore other options.

Let me know yours and SETI's thoughts on this. Thanks!

Reply:
Yeah, this action is encapsulating secret-scanning as part of it and could even replace it in the future if our experiment goes well, so same reason for `depth: 0` :)

Interesting that there's already some sort of solution in place. One possibility would be doing the checkout as part of our action itself, but it might be tricky to get it right (we can look at the source for `actions/checkout` to have a rough idea of how tricky it is). 

In general I would like to say 'yes' to this, but I wouldn't want tackle it right now. We want to validate this approach of having a single action encompassing many checks with teams first, so I'd keep it simple as a first iteration and them try to get something more sophisticated working in the future.

Reply:
Ok, makes sense, let's keep it like this for now then.

Thanks!